src/pages/sources/components/AddSourceModal.tsx

```diff
@@ -155,24 +162,32 @@ const AddSourceModal: React.FC<AddSourceModalProps> = ({ source, type, onClose,
                 </FormGroup>
               </>
             ) : (
-              <FormGroup label="IP address or hostname" isRequired fieldId="hosts">
-                <TextInput
-                  value={getValue('hosts')}
-                  onChange={(_ev, val) => setValue('hosts', val)}
-                  isRequired
-                  id="source-hosts"
-                  name="hosts"
-                />
-                <HelperText>Enter an IP address or hostname (Default port is 443)</HelperText>
-              </FormGroup>
+              <>
+                <FormGroup label="IP address or hostname" isRequired fieldId="hosts">
+                  <TextInput
+                    value={getValue('hosts')}
+                    onChange={(_ev, val) => setValue('hosts', val)}
+                    isRequired
+                    id="source-hosts"
+                    name="hosts"
+                  />
+                  <HelperText>Enter an IP address or hostname</HelperText>
+                </FormGroup>
+                <FormGroup label="Port" fieldId="port">
+                  <TextInput
+                    value={getValue('port')}
+                    placeholder="Optional"
+                    type="text"
+                    id="source-port"
+                    name="port"
+                    onChange={ev => {
+                      setValue('port', (ev.target as HTMLInputElement).value);
+                    }}
+                  />
+                  <HelperText>Default port is 443</HelperText>
```

Comment:
For what source type this 443 (HTTPS) refers to? I mean, for network (SSH) the default port is 22.

Reply:
yes, network default port is specified somewhere else.

Reply:
values['port'] || (isNetwork ? '22' : '443')